legend-engine-xt-analytics-mapping-api/src/main/java/org/finos/legend/engine/analytics/mapping/model/coverage/api/ModelCoverageAnalytics.java

```diff
@@ -0,0 +1,165 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+package org.finos.legend.engine.analytics.mapping.model.coverage.api;
+
+import io.opentracing.Scope;
+import io.opentracing.util.GlobalTracer;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import org.eclipse.collections.api.RichIterable;
+import org.eclipse.collections.api.list.MutableList;
+import org.finos.legend.engine.language.pure.compiler.toPureGraph.PureModel;
+import org.finos.legend.engine.language.pure.modelManager.ModelManager;
+import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.domain.Multiplicity;
+import org.finos.legend.engine.shared.core.api.result.ManageConstantResult;
+import org.finos.legend.engine.shared.core.kerberos.ProfileManagerHelper;
+import org.finos.legend.engine.shared.core.operational.errorManagement.ExceptionTool;
+import org.finos.legend.engine.shared.core.operational.http.InflateInterceptor;
+import org.finos.legend.engine.shared.core.operational.logs.LoggingEventType;
+import org.finos.legend.pure.generated.platform_pure_corefunctions_multiplicity;
+import org.finos.legend.pure.m3.coreinstance.meta.pure.mapping.Mapping;
+import org.finos.legend.pure.runtime.java.compiled.execution.CompiledExecutionSupport;
+import org.pac4j.core.profile.CommonProfile;
+import org.pac4j.core.profile.ProfileManager;
+import org.pac4j.jax.rs.annotations.Pac4JProfileManager;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.finos.legend.pure.generated.core_analytics_mapping_modelCoverageAnalytics;
+import org.finos.legend.pure.generated.Root_meta_analytics_mapping_modelCoverage_MappedEntity;
+import org.finos.legend.pure.generated.Root_meta_analytics_mapping_modelCoverage_MappedProperty;
+import org.finos.legend.pure.generated.Root_meta_analytics_mapping_modelCoverage_EntityMappedProperty;
+import org.finos.legend.pure.generated.Root_meta_analytics_mapping_modelCoverage_EnumMappedProperty;
+
+
+@Api(tags = "Analytics - Model")
+@Path("pure/v1/analytics/mapping")
+public class ModelCoverageAnalytics
+{
+    private final ModelManager modelManager;
+
+    public ModelCoverageAnalytics(ModelManager modelManager)
+    {
+        this.modelManager = modelManager;
+    }
+
+    @POST
+    @Path("modelCoverage")
+    @ApiOperation(value = "Analyze the input pure mapping to generate information about mapped classes and mapped properties of each class")
+    @Consumes({MediaType.TEXT_PLAIN, InflateInterceptor.APPLICATION_ZLIB})
```

Comment:
Also, I forgot, this endpoint should not accept `TEXT_PLAIN`, probably `APPLICATION_JSON`